[dynamo] graph break on nn.Parameter construction

davidberard98 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #103262

Fixes #99569

nn.Parameter construction appears to run into FakeTensor / tracing issues during AOT Autograd. We could try to fix this; but nn.Parameter construction _inside_ the compiled region isn't a common scenario, so it's reasonable to just graph break on nn.Parameter construction.

For reference, see #99569 for the errors/issues that appear from tracing through nn.Parameter construction with AOT Autograd.

cc @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @ipiszy @aakhundov

davidberard98 committed:
[dynamo] graph break on nn.Parameter construction

nn.Parameter construction appears to run into FakeTensor / tracing issues during AOT Autograd. We could try to fix this; but nn.Parameter construction _inside_ the compiled region isn't a common scenario, so it's reasonable to just graph break on nn.Parameter construction.

[ghstack-poisoned]

williamwen42 reviewed (commented):
This error happens happens in 14k github models (https://github.com/pytorch/pytorch/issues/102989 for the repro). We can graph break for now (and the repro in the linked issue will pass) but we should try to fix this for real at some point.

davidberard98 commented:
@pytorchbot merge -r

davidberard98 committed:
Update on "[dynamo] graph break on nn.Parameter construction"


Fixes #99569

nn.Parameter construction appears to run into FakeTensor / tracing issues during AOT Autograd. We could try to fix this; but nn.Parameter construction _inside_ the compiled region isn't a common scenario, so it's reasonable to just graph break on nn.Parameter construction.

For reference, see #99569 for the errors/issues that appear from tracing through nn.Parameter construction with AOT Autograd.

cc voznesenskym penguinwu anijain2305 EikanWang jgong5 Guobing-Chen XiaobingSuper zhuhaozhe blzheng Xia-Weiwen wenzhe-nrv jiayisunx ipiszy aakhundov

[ghstack-poisoned]

davidberard98 commented:
@pytorch merge
